src/Compilers/VisualBasic/Portable/Binding/InitializerSemanticModel.vb

```diff
@@ -106,7 +114,8 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                         Dim boundInitializers = ArrayBuilder(Of BoundInitializer).GetInstance
                         If initializer IsNot Nothing Then
                             ' bind const and non const field initializers the same to get a bound expression back and not a constant value.
-                            binder.BindFieldInitializer(ImmutableArray.Create(Of FieldSymbol)(fieldSymbol), initializer, boundInitializers, diagnostics, bindingForSemanticModel:=True)
+                            Dim fields = ImmutableArray.CreateRange(GetInitializedFieldsOrProperties(binder).Cast(Of FieldSymbol))
```

Comment:
>ImmutableArray.CreateRange(GetInitializedFieldsOrProperties(binder).Cast(Of FieldSymbol)) [](start = 41, length = 89)

```binder.AdditionalContainingMembers.Cast(Of FieldSymbol).Insert(0, fieldSymbol)```?

Reply:
Actually that does not work as Cast invocation turns it into an IEnumerable, which does not have an Insert method. I am going to leave these as is for now.

---
In reply to: [149830887](https://github.com/dotnet/roslyn/pull/23077#discussion_r149830887) [](ancestors = 149830887)